external/fv3fit/fv3fit/keras/_models/models.py

```diff
@@ -99,13 +106,15 @@ def __init__(
         self._spectral_normalization = hyperparameters.spectral_normalization
         self._gaussian_noise = hyperparameters.gaussian_noise
         self._nonnegative_outputs = hyperparameters.nonnegative_outputs
+        # TODO: remove internal sample dim name once sample dim is hardcoded everywhere
+        self._fv3fit_sample_dim = f"_sample_fv3fit"
```

Comment:
Instead of changing SAMPLE_DIM_NAME to be an instance variable, you should be able to update SAMPLE_DIM_NAME to be "_sample_fv3fit".